utils/python/CIME/XML/env_archive.py

```diff
@@ -22,7 +22,7 @@ def __init__(self, case_root=None, infile="env_archive.xml"):
         else:
             fullpath = os.path.join(case_root, infile)
         GenericXML.__init__(self, fullpath)
-        if not os.path.isfile(infile):
+        if not os.path.isfile(fullpath):
```

Comment:
use of relative path infile instead of fullpath was causing the env_archive object to be appended each time init was called.